keps/sig-node/20190925-enable-seccomp-by-default.md

```diff
@@ -0,0 +1,335 @@
+
+The support is based on the downstream dependencies, therefore Kubernetes changes are not _required_. However, some changes may be improve its usability as pointed out on the [Implementation Details](#implementation-details) section.
+
+
+### 2. Built-in profiles
+The table below shows what built-in profiles and the two supported ways to create user-defined profiles.
+
+| Profile Name 	| Description 	| Status 	| Requires Audit Support 	| Fallback profile 	|
+|-------------------------	|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|-------------------------------------------	|-----------------------------------------	|------------------	|
+| `runtime/default` 	| The default container runtime. Syscalls outside allowed list are blocked. 	| Unchanged 	| No 	| N/A 	|
+| `runtime/default-audit` 	| Allows the same syscalls as `runtime/default`, but logs all violations. 	| New 	| Yes 	| `unconfined` 	|
+| `runtime/audit-verbose` 	| Remove all whitelisted calls, logging every time any system calls are used. Useful for creating new profiles based on the execution of a container. 	| New 	| Yes 	| `unconfined` 	|
+| `custom/<profile-name>` 	| User defined profiles based off the new `SeccompProfile` Kind. 	| New 	| Only when `SCPM_ACT_LOG` is being used 	| `unconfined` 	|
+| `localhost/<path>` 	| User defined profile as a file on the node located at <seccomp_root>/<path>, where <seccomp_root> is defined via the  --seccomp-profile-root flag on the Kubelet. _Note that the user is responsible for physically synchronising the profile files across all nodes._ 	| Unchanged 	| Only when ` SCPM_ACT_LOG` is being used 	| `unconfined` 	|
+| `docker/default` 	| The Docker default seccomp profile is used. Deprecated as of Kubernetes 1.11. Use  `runtime/default` instead. 	| Unchanged, Deprecated 	| No 	| N/A 	|
+| `unconfined` 	| Seccomp is not applied to the container processes (the current default in Kubernetes), if no alternative is provided. Usage should be disencouraged. 	| Unchanged 	| No 	| N/A 	|
+
+
+### 3. New Kind for custom profiles
+
+Promote seccomp profiles to first-class citizen as a new *cluster-level* resource, represented by a new `SeccompProfile` API resource. 
+The new API resource would remove the existing overhead of creating and synchronising files across all new and existing nodes. Allowing users to maintain custom profiles by interacting with the API server.
+
+More details on types needed and usage on the [Implementation Details](#implementation-details) section.
+
+ 
+### 4. Set `runtime/default-audit` as the default profile
```

Comment:
Is the end-goal still to enable the enforcing profile by default? If that's the case, I wonder if it's better to start by making this available, and then establish a roleout plan for getting to the end state?

Reply:
Yes, you are absolutely correct, the end-goal is to achieve enforceable seccomp by default. I believe first the audit mode needs to land, then we probably can start creating profiles for the internal Kubernetes components. In that process we would create tools to help users to do the same for their workloads.

Reply:
@tallclair In the Graduation Criteria I added that "set as default" would only make it by GA. This would be the mid term solution for user workloads. 

I will take a look at enforcing seccomp in the components shipped by default in parallel to this.